File: android/app/src/main/java/com/candy/handyman/data/remote/dto/ServiceDto.kt
package com.candy.handyman.data.remote.dto

data class ServiceDto(
    val id: String,
    val title: String,
    val description: String,
    val pricingType: String,
    val price: Double,
    val unit: String,
    val status: String,
    val viewCount: Int,
    val categoryId: String,
    val categoryName: String? = null,
    val handyman: HandymanDto? = null,
    val media: List<ServiceMediaDto> = emptyList(),
    val createdAt: String
)

data class ServiceMediaDto(
    val id: String,
    val mediaType: String,
    val mediaUrl: String,
    val coverUrl: String?,
    val sortOrder: Int
)

data class HandymanDto(
    val id: String,
    val nickname: String,
    val avatar: String?,
    val yearsOfExperience: Int,
    val averageRating: Double,
    val totalReviews: Int,
    val isVerified: Boolean
)

data class CategoryDto(
    val id: String,
    val name: String,
    val icon: String?,
    val parentId: String?,
    val sortOrder: Int,
    val children: List<CategoryDto> = emptyList()
)

data class CreateServiceDto(
    val title: String,
    val description: String,
    val categoryId: String,
    val pricingType: String,
    val price: Double,
    val unit: String
)

File: src/CandyHandyman.Application/DTOs/ServiceDto.cs
using CandyHandyman.Core.Enums;

namespace CandyHandyman.Application.DTOs;

public class ServiceDto
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public PricingType PricingType { get; set; }
    public decimal Price { get; set; }
    public string Unit { get; set; } = string.Empty;
    public ServiceStatus Status { get; set; }
    public int ViewCount { get; set; }
    public Guid CategoryId { get; set; }
    public string CategoryName { get; set; } = string.Empty;
    public HandymanDto Handyman { get; set; } = null!;
    public List<ServiceMediaDto> Media { get; set; } = new();
    public DateTime CreatedAt { get; set; }
}

public class ServiceMediaDto
{
    public Guid Id { get; set; }
    public MediaType MediaType { get; set; }
    public string MediaUrl { get; set; } = string.Empty;
    public string? CoverUrl { get; set; }
    public int SortOrder { get; set; }
}

public class CreateServiceDto
{
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public Guid CategoryId { get; set; }
    public PricingType PricingType { get; set; }
    public decimal Price { get; set; }
    public string Unit { get; set; } = string.Empty;
}

public class HandymanDto
{
    public Guid Id { get; set; }
    public string Nickname { get; set; } = string.Empty;
    public string? Avatar { get; set; }
    public int YearsOfExperience { get; set; }
    public decimal AverageRating { get; set; }
    public int TotalReviews { get; set; }
    public bool IsVerified { get; set; }
}

public class CategoryDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Icon { get; set; }
    public Guid? ParentId { get; set; }
    public int SortOrder { get; set; }
    public List<CategoryDto> Children { get; set; } = new();
}